specification/apimanagement/resource-manager/Microsoft.ApiManagement/preview/2018-06-01-preview/apimdiagnostics.json

```diff
@@ -0,0 +1,533 @@
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "./apimanagement.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ServiceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DiagnosticIdParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/IfMatchRequiredParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The Diagnostic was successfully deleted."
+          },
+          "204": {
+            "description": "The Diagnostic was successfully deleted."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./apimanagement.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DiagnosticCollection": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiagnosticContract"
+          },
+          "description": "Page values."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "Next page link if any."
+        }
+      },
+      "description": "Paged Diagnostic list representation."
+    },
+    "DiagnosticContract": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/DiagnosticContractProperties",
+          "description": "Diagnostic entity contract properties."
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "./apimanagement.json#/definitions/Resource"
+        }
+      ],
+      "description": "Diagnostic details."
+    },
+    "SamplingSettings": {
+      "properties": {
+        "samplingType": {
+          "type": "string",
+          "description": "Sampling type.",
+          "enum": [
+            "fixed"
+          ],
+          "x-ms-enum": {
+            "name": "SamplingType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "fixed",
+                "description": "Fixed-rate sampling."
+              }
+            ]
+          }
+        },
+        "percentage": {
+          "type": "number",
+          "format": "double",
+          "minimum": 0,
+          "maximum": 100,
+          "description": "Rate of sampling for fixed-rate sampling."
+        }
+      },
+      "description": "Sampling settings for Diagnostic."
+    },
+    "PipelineDiagnosticSettings": {
+      "properties": {
+        "request": {
+          "allOf": [
```

Comment:
>allOf [](start = 11, length = 5)

why "allOf" ... and why not a simple $ref
  ` request : {
     "$ref": "#/definitions/HttpMessageDiagnostic",
      "description: ""
    }
 `

Reply:
fixed